phase4-profile-bdew/src/main/java/com/helger/phase4/profile/bdew/BDEWCompatibilityValidator.java

```diff
@@ -509,14 +511,36 @@ public void validateUserMessage (@Nonnull final Ebms3UserMessage aUserMsg, @Nonn
       if (aFrom != null)
       {
         if (aFrom.getPartyIdCount () > 1)
+        {
           aErrorList.add (_createError ("PartyInfo/From must contain no more than one PartyID"));
+        }
+
+        if (StringHelper.hasNoText (aFrom.getRole ()) || !aFrom.getRole ().equals (CAS4.DEFAULT_INITIATOR_URL))
+        {
+          aErrorList.add (_createError ("PartyInfo/From/Role must be set to '" + CAS4.DEFAULT_INITIATOR_URL + "'"));
+        }
+      }
+      else
+      {
+        aErrorList.add (_createError ("PartyInfo/From is missing"));
       }
 
       final Ebms3To aTo = aUserMsg.getPartyInfo ().getTo ();
       if (aTo != null)
       {
         if (aTo.getPartyIdCount () > 1)
+        {
           aErrorList.add (_createError ("PartyInfo/To must contain no more than one PartyID"));
+        }
+
+        if (StringHelper.hasNoText (aTo.getRole ()) || !aTo.getRole ().equals (CAS4.DEFAULT_RESPONDER_URL))
```

Comment:
That's the same as if (!CAS4.DEFAULT_RESPONDER_URL.equals (aFrom.getRole ())) right?